DataFormats/CTPPSDigi/interface/CTPPSPixelDataError.h

```diff
@@ -36,16 +35,22 @@ class CTPPSPixelDataError {
   /// the fedId where the error occured    
   inline int fedId() const {return fedId_;} 	        
   /// the error message to be displayed with the error	
-  inline const std::string & errorMessage() const {return errorMessage_;}    
-  
+  inline const std::string & errorMessage() const {return errorMessage_;}   
+
+
 private:
-	
-    uint64_t errorWord64_;
-    uint32_t errorWord32_; 
-    int errorType_;
-    int fedId_;
-    std::string errorMessage_;
-    
+
+  static const std::string errorMessages[];
+  
+  /// create an error message based on errorType
+  void setMessage();
+  uint64_t errorWord64_;
+  uint32_t errorWord32_; 
+  int errorType_;
+  int fedId_;
+  std::string errorMessage_;

```

Comment:
going the way of using the static array implied that we do not store the string anymore and access the array directly.

Reply:
This would mean removing errorMessage_, errorMessage() and setMessage(), keeping only the array. That's what you mean?

Reply:
- removing errorMessage_
- removing setMessage()
- reimplement errorMessage to return errorMessages[x]

Reply:
(probably) done